core/src/banking_stage.rs

```diff
@@ -1435,6 +1442,30 @@ impl BankingStage {
         }
     }
 
+    /// To commit transaction cost to cost_tracker if it was executed successfully;
+    /// Otherwise cancel it from being committed, therefore prevents cost_tracker
+    /// being inflated with unsuccessfully executed transactions.
+    fn commit_or_cancel_transaction_cost<'a>(
+        transactions: impl Iterator<Item = &'a SanitizedTransaction>,
+        transaction_results: impl Iterator<Item = &'a transaction::Result<()>>,
+        retryable_transaction_indexes: &[usize],
+        qos_service: &QosService,
+        bank: &Arc<Bank>,
+    ) {
+        transactions
+            .zip(transaction_results)
+            .enumerate()
+            .for_each(|(index, (tx, result))| {
+                if result.is_ok() && retryable_transaction_indexes.contains(&index) {
```

Comment:
if we pass in `transactions` and `transaction_resutls` as slices and instead iterate on `retryable_transaction_indexes` here, we can safely index into the slices and avoid the `.contains()` O(N) lookup

Reply:
Agreed. @taozhu-chicago is fixing a bug in this area anyway, and we'll likely iterate over `transaction_qos_results.zip(transaction_execution_results)` or something, and won't need the retryable_indices at all.